### PR TITLE
Hotfix/nav and styling bugs

### LINKED ISSUE #201: Truncating line graphs to integer values has reappeared in Dev environment.
This was fixed in previous issue, and is fixed in Prod.  
It appears to have reappeared in Development environment, where the MW values are truncated (or rounded) to integer values.  Openinng this issue to ensure it is not lost when the next move to production occurs. 

See here: 
image

### PR DESCRIPTION
**Pull request by braddf**
# Pull Request

Various bugs/styling quirks as per commit messages, and as seen by @dantravers and myself during manual testing.

Fixes #201 

## How Has This Been Tested?

Visually, and will hopefully be on a Preview branch for the team to double check.

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
Screenshot 2022-10-25 at 12 00 33
 screen shot of new app

**Comment by braddf** [comment]
@rachel-labri-tipton Perfect, no by all means jump on here and keep things in one place 👍 
I'm just trying to fix one of the tests on the API stuff for auth so if you want to work on these that would be great!

**Comment by rachel-labri-tipton** [comment]
morning @dantravers, just following up on the slack action I missed yesterday. I added time on this branch to the date earlier this week. the previews aren't currently working because of updates on dev. but the date's there, and it's the same one we use on the tooltip, so 24h clock.

**Comment by rachel-labri-tipton** [comment]
this last commit was just commenting out some code after the merge

**Comment by peterdudfield** [comment]
Comments from meeting
- space out legend, space between the columns
- Add env variable to turn on/off 4 hour forecast
-

**Comment by rachel-labri-tipton** [comment]
hey @braddf, I'm getting the following error when I try to commit my code. 

Screenshot 2022-11-14 at 16 03 52

I don't think we're able to use the useSWR hook as part of a conditional statement within the context of the `useGetGspData` component.

**Comment by rachel-labri-tipton** [comment]
hi @braddf, having looked at @peterdudfield's comment from yesterday, the suggestion was to create an `issue` for the 4hforecast environment variable/ global state variable. given that it wasn't the quickest of fixes for me yesterday, I'd like to go ahead and make it an issue. 

to that end, I've commented out the code we wrote yesterday and wanted to ask if we could merge this branch into development, because we have attended to the hotfixes on this branch. 

I'll then make the following issues: 
- Add env variable to turn on/off 4 hour forecast 
- bug: Remove delta code that slipped in on accident

**Review by rachel-labri-tipton — APPROVED** [review]

**Comment by braddf** [comment]
Hey @rachel-labri-tipton - I get where you're coming from. 

In terms of commits, I think stopping before the whole toggle solution was implemented is wise. I have to remind myself to commit things and try to chunk up work more sensibly instead of charging through onto the next thing ASAP 😁 

But in these cases, although I don't think we need to do this for this time, in general if we think we have gone outside of the scope of an Issue + PR, then that code shouldn't be included at all, and be moved to said other, new branch with a new issue that fits the code better.

For this time, I think what you have here is okay, and the commits should be sufficient to separate what we're trying to achieve here so far, so next I would say let's re-jig this PR, rename it to encompass both the UX fixes and this Dev/Prod toggle, and then it will not be at risk of leaving `development` (which is really our `staging` branch now) in a state that we don't want - i.e. without a 4hr forecast, even if temporarily. This way, the PR can close the original issue(s) and the new "Add env" issue, and a commit removing the erroneous Delta code is also fine without an issue as nothing has been merged yet.

That sound okay?

**Comment by rachel-labri-tipton** [comment]
@braddf, thank you thank you for your thoughts. this sounds sensible and good to me. I'll remove the delta stuff now and have a look at the 4h toggle again tomorrow. I wanted to give some time to delta view this afternoon.

File: apps/nowcasting-app/components/charts/pv-remix-chart.tsx [hunk 1]
```diff
@@ -45,6 +46,8 @@ const LegendItem: FC<{
 };
 
 const PvRemixChart: FC<{ date?: string; className?: string }> = ({ className }) => {
+  // const show4hView = process.env.NEXT_PUBLIC_4H_VIEW === "true";
```
Comment (braddf) [inline apps/nowcasting-app/components/charts/pv-remix-chart.tsx:49]: Remove commented code

File: apps/nowcasting-app/components/layout/header/profile-dropdown.tsx [hunk 2]
```diff
@@ -13,6 +13,14 @@ interface IProfileDropDown {}
 
 const ProfileDropDown = ({}: IProfileDropDown) => {
   const { user } = useUser();
+  const [show4hView, setShow4hView] = useGlobalState("show4hView");
+  const toggle4HourForecastVisibility = () => {
+    if (show4hView === false) {
```
Comment (braddf) [inline apps/nowcasting-app/components/layout/header/profile-dropdown.tsx:18]: For a simple toggle function like this, can be shortened just to something like: `() => setShow4hView(!show4hView)` and can either go into a wrapper function like you have here, or if it's simple can just go inline in the onClick handler.

File: apps/nowcasting-app/components/layout/header/profile-dropdown.tsx [hunk 3]
```diff
@@ -83,9 +91,24 @@ const ProfileDropDown = ({}: IProfileDropDown) => {
...
+                className={classNames(
+                  active ? "bg-gray-100" : "",
+                  "block px-4 py-2 text-sm text-gray-700"
+                )}
+              >
+                <button
+                  onClick={toggle4HourForecastVisibility}
+                  className="ml-3 text-sx  font-medium text-ocf-black-300 dark:text-gray-300"
+                >
+                  {show4hView ? "Hide 4-hour forecast" : "Show 4-hour forecast"}
```
Comment (braddf) [inline apps/nowcasting-app/components/layout/header/profile-dropdown.tsx:107]: Also can be shortened for nice, succinct and, most importantly, DRY code: 
```
{`${show4hView ? "Hide" : "Show"} 4-hour forecast`}
```

File: apps/nowcasting-app/pages/index.tsx [hunk 4]
```diff
@@ -19,14 +20,14 @@ export default function Home() {
...
-          className={`relative float-right h-full ${currentView(VIEWS.FORECAST) ? "" : "hidden"}`}
+          className={`relative float-right h-full ${currentView(VIEWS.FORECAST)}`}
           style={{ width: "56%" }}
         >
           <PvLatestMap />
         </div>
 
         <SideLayout>
-          <PvRemixChart className={currentView(VIEWS.FORECAST) ? "" : "hidden"} />
+          <PvRemixChart className={`${currentView(VIEWS.FORECAST)}`} />
```
Comment (braddf) [inline apps/nowcasting-app/pages/index.tsx:30]: If no other characters in this string literal, doesn't need the extra brackets and back ticks I think?

File: apps/nowcasting-app/components/map/pvLatest.tsx [hunk 5]
```diff
@@ -199,15 +199,15 @@ const PvLatestMap = () => {
...
+      <ButtonGroup rightString={formatISODateStringHuman(selectedISOTime || " ")} />
     </LoadStateMap>
   ) : (
     <Map
       loadDataOverlay={addFCData}
       updateData={{ newData: !!initForecastData, updateMapData }}
       controlOverlay={(map: { current?: mapboxgl.Map }) => (
         <>
-          <ButtonGroup rightString={formatISODateStringHuman(selectedISOTime || "")} />
+          <ButtonGroup rightString={formatISODateStringHuman(selectedISOTime || " ")} />
```
Comment (braddf) [inline apps/nowcasting-app/components/map/pvLatest.tsx:210]: Are these new spaces intentional here and above on line 202? No worries if they are necessary, just checking wasn't a weird linting thing

Comment (rachel-labri-tipton) [reply apps/nowcasting-app/components/map/pvLatest.tsx:210]: this wasn't intentional

File: apps/nowcasting-app/components/layout/header/profile-dropdown.tsx (hunk 2 above)
Comment (rachel-labri-tipton) [reply apps/nowcasting-app/components/layout/header/profile-dropdown.tsx:18]: thanks for showing me how to make these simpler!

File: apps/nowcasting-app/pages/index.tsx (hunk 4 above)
Comment (rachel-labri-tipton) [reply apps/nowcasting-app/pages/index.tsx:30]: typescript is telling me off for taking the back ticks out. that might be why they're there.

**Comment by rachel-labri-tipton** [comment]
hey @dantravers, so I think the idea for this feature is not to make it available to a user. I totally agree they shouldn't have that power/access. it's one piece of code to comment out when we deploy to production instead of commenting out all of the 4h-forecast code. at least that's how I understand it.

**Comment by dantravers** [comment]
> hey @dantravers, so I think the idea for this feature is not to make it available to a user. I totally agree they shouldn't have that power/access. it's one piece of code to comment out when we deploy to production instead of commenting out all of the 4h-forecast code. at least that's how I understand it.

Gotcha - thanks!

**Pull request merged**